CMS-Enterprise/mint-app
# [MINT-3235] Get milestones and solutions by relation loaders and helper methods

# MINT-3235

## Description

1. This is a follow up PR to https://github.com/CMS-Enterprise/mint-app/pull/1557
2. Unit tests for getting solutions and milestones
3. Updated mapping function to allow a custom maptype

## How to test this change

1.  Use postman to create a common milestone with common solutions
2. Use the query to return milestones and their solutions,
     a. ensure that the new common solutions are returned
    b. ensure the nested milestone returned is that milestone

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
